src/qililab/instruments/quantum_machines/quantum_machines_cluster.py

```diff
@@ -0,0 +1,354 @@
+# Copyright 2023 Qilimanjaro Quantum Tech
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Quantum Machines Manager class."""
+import os
+from dataclasses import dataclass
+from typing import Any, Dict
+
+import numpy as np
+from qm import DictQuaConfig, QuantumMachine, QuantumMachinesManager, SimulationConfig
+from qm.jobs.running_qm_job import RunningQmJob
+from qm.octave import QmOctaveConfig
+from qm.qua import Program
+
+from qililab.instruments.instrument import Instrument
+from qililab.instruments.utils import InstrumentFactory
+from qililab.typings import InstrumentName, Parameter, QMMDriver
+from qililab.utils import merge_dictionaries
+
+
+@InstrumentFactory.register
+class QuantumMachinesCluster(Instrument):
+    """Class defining the Qililab Quantum Machines Cluster instrument in Qililab.
+
+    This class allows Qililab control and communication with an instance of the
+    Quantum Machines Manager, which is the central class to interact with Quantum Machines instruments.
+    The manager is responsible, through the use of QUA sequences, of setting the Quantum Machines instruments
+    in the right manner for Quantum Control.
+
+    Args:
+        name (InstrumentName): Name of the Instrument.
+        device (QMMDriver): Instance of the Quantum Machines Manager Driver class.
+        settings (QMMSettings): Settings of the instrument.
```

Comment:
The following attributes are missing in the docstring: 
    
    qmm: QuantumMachinesManager
    qm: QuantumMachine
    config: DictQuaConfig
    octave_config: QmOctaveConfig | None = None

Reply:
Should we make everything private? None of these should be accessed by the user directly.

Reply:
Maybe the `config` could be exposed by a get-only property.

Reply:
Those both are good ideas.

Reply:
Done. I didn't create the config property because it is not needed anywhere.